st2common/st2common/models/api/auth.py

```diff
@@ -92,3 +94,49 @@ def to_model(cls, instance):
 
         model = cls.model(user=user, token=token, expiry=expiry)
         return model
+
+
+class ApiKeyAPI(BaseAPI):
+    model = ApiKeyDB
+    schema = {
+        "title": "ApiKey",
+        "type": "object",
+        "properties": {
+            "id": {
+                "type": "string"
+            },
+            "user": {
+                "type": ["string", "null"],
+                "default": ""
+            },
+            "key": {
+                "type": ["string", "null"]
+            },
+            "metadata": {
+                "type": ["object", "null"]
+            }
+        },
+        "additionalProperties": False
+    }
+
+    @classmethod
+    def from_model(cls, model, mask_secrets=False):
+        doc = cls._from_model(model=model, mask_secrets=mask_secrets)
+        attrs = {attr: value for attr, value in six.iteritems(doc) if value is not None}
+        # switcharoo. key_hash is returned as key and likely masked.
+        attrs['key'] = attrs.pop('key_hash', None)
+
+        return cls(**attrs)
+
+    @classmethod
+    def to_model(cls, instance):
+        user = str(instance.user) if instance.user else None
+        key = str(instance.key) if instance.key else None
+        metadata = getattr(instance, 'metadata', {})
+
+        # The api_key is never really stored in the DB. It is assumed
```

Comment:
Assuming it is not too messy to handle in the controller I can make that change.

What is passed into POST @ /apikeys? For now that method takes ApiKeyAPI. Should it be ApiKetCreateSpec since key_hash cannot be input? I suppose I can entirely ignore key_hash for POST still it does not particularly help with consistency.